tests/test_iam/test_iam.py

```diff
@@ -2879,3 +2880,450 @@ def test_delete_role_with_instance_profiles_present():
     role_names = [role["RoleName"] for role in iam.list_roles()["Roles"]]
     assert "Role1" in role_names
     assert "Role2" not in role_names
+
+
+@mock_iam
+def test_delete_account_password_policy_errors():
+    client = boto3.client("iam", region_name="us-east-1")
+
+    client.delete_account_password_policy.when.called_with().should.throw(
+        ClientError, "The account policy with name PasswordPolicy cannot be found."
+    )
+
+
+@mock_iam
+def test_role_list_config_discovered_resources():
```

Comment:
Add a test condition for filtering on name and ID. This should reveal an issue with IAM role names and IDs being different.

Reply:
good catch; refactored the backend a little bit to return the role id and name separately